vic/drivers/image/src/vic_restore.c

```diff
@@ -29,39 +29,987 @@
+                                            "AnnualNPPPrev",
+                                            d4start, d4count, dvar);
+                for (i = 0; i < local_domain.ncells_active; i++) {
+                    v = veg_con_map[i].vidx[m];
+                    if (v >= 0) {
+                        all_vars[i].veg_var[v][k].AnnualNPPPrev = dvar[i];
+                    }
+                }
+            }
+        }
+
+        // litter carbon: tmpval = cell[veg][band].CLitter;
+        for (m = 0; m < options.NVEGTYPES; m++) {
+            d4start[0] = m;
+            for (k = 0; k < options.SNOW_BAND; k++) {
+                d4start[1] = k;
+                get_scatter_nc_field_double(filenames.init_state,
+                                            "CLitter",
+                                            d4start, d4count, dvar);
+                for (i = 0; i < local_domain.ncells_active; i++) {
+                    v = veg_con_map[i].vidx[m];
+                    if (v >= 0) {
+                        all_vars[i].cell[v][k].CLitter = dvar[i];
+                    }
+                }
+            }
+        }
+
+        // intermediate carbon: tmpval = tmpval = cell[veg][band].CInter;
+        for (m = 0; m < options.NVEGTYPES; m++) {
+            d4start[0] = m;
+            for (k = 0; k < options.SNOW_BAND; k++) {
+                d4start[1] = k;
+                get_scatter_nc_field_double(filenames.init_state,
+                                            "CInter",
+                                            d4start, d4count, dvar);
+                for (i = 0; i < local_domain.ncells_active; i++) {
+                    v = veg_con_map[i].vidx[m];
+                    if (v >= 0) {
+                        all_vars[i].cell[v][k].CInter = dvar[i];
+                    }
+                }
+            }
+        }
+
+        // slow carbon: tmpval = cell[veg][band].CSlow;
+        for (m = 0; m < options.NVEGTYPES; m++) {
+            d4start[0] = m;
+            for (k = 0; k < options.SNOW_BAND; k++) {
+                d4start[1] = k;
+                get_scatter_nc_field_double(filenames.init_state,
+                                            "CSlow",
+                                            d4start, d4count, dvar);
+                for (i = 0; i < local_domain.ncells_active; i++) {
+                    v = veg_con_map[i].vidx[m];
+                    if (v >= 0) {
+                        all_vars[i].cell[v][k].CSlow = dvar[i];
+                    }
+                }
+            }
+        }
+    }
+
+    // snow age: snow[veg][band].last_snow
+    for (m = 0; m < options.NVEGTYPES; m++) {
+        d4start[0] = m;
+        for (k = 0; k < options.SNOW_BAND; k++) {
+            d4start[1] = k;
+            get_scatter_nc_field_int(filenames.init_state,
+                                     "Snow_age",
+                                     d4start, d4count, ivar);
```

Comment:
So you want me to make this change as part of this (and [UW-Hydro/tonic#43](https://github.com/UW-Hydro/tonic/pull/43)) PR?

I'd prefer to make that a separate issue, since it's going to affect the classic state file i/o and the snow_utility.c and lakes.eb.c code too.  The purpose of this PR was simply to bring the image driver up to the same functionality as the classic driver in terms of state file i/o.